src/coord/src/coord.rs

```diff
@@ -598,10 +603,16 @@ impl Coordinator {
     ) {
         match message {
             WorkerFeedback::FrontierUppers(updates) => {
-                for (name, changes) in updates {
-                    self.update_upper(&name, changes);
+                let mut durability_updates = Vec::new();
+                for feedback in updates {
+                    self.update_upper(feedback, &mut durability_updates);
                 }
                 self.maintenance().await;
+                if !durability_updates.is_empty() {
+                    self.broadcast(SequencedCommand::DurabilityFrontierUpdates(
+                        durability_updates,
+                    ));
+                }
             }
         }
     }
```

Comment:
I don't understand what is going on here. Previously, a thing which provided upper frontiers responded, and we applied some logic, and .. I could sort of just see from the code what was happening. Now, .. there are durability things that are formed and this results in an automatic broadcast to all workers. I don't understand why these are related.